src/ray/core_worker/store_provider/memory_store_provider.cc

```diff
@@ -18,7 +18,13 @@ CoreWorkerMemoryStoreProvider::CoreWorkerMemoryStoreProvider(
 
 Status CoreWorkerMemoryStoreProvider::Put(const RayObject &object,
                                           const ObjectID &object_id) {
-  return store_->Put(object_id, object);
+  auto status = store_->Put(object_id, object);
+  if (status.IsObjectExists()) {
+    // object already exists in store, treat it as ok.
```

Comment:
```suggestion
    // Object already exists in store, treat it as ok.
```

Please add a comment explaining why we treat this as ok.